keanu-project/src/main/java/io/improbable/keanu/algorithms/particlefiltering/Particle.java

```diff
@@ -0,0 +1,83 @@
+package io.improbable.keanu.algorithms.particlefiltering;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.ProbabilityCalculator;
+import io.improbable.keanu.vertices.Vertex;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a Particle used in a Particle Filter.
+ * A Particle can be thought of as a particular instance of the network with state and an associated probability.
+ */
+public class Particle {
+
+    private Map<Vertex, Object> latentVertices = new HashMap<>();
+    private List<Vertex> observedVertices = new ArrayList<>();
+    private double sumLogPOfSubgraph = 1.0;
+
+    public Map<Vertex, Object> getLatentVertices() {
+        return latentVertices;
+    }
+
+    /**
+     * @return log probability of the subgraph occuring in it's current state
+     */
+    public double logProb() {
+        return sumLogPOfSubgraph;
+    }
+
+    /**
+     * @return probability of the subgraph occuring in it's current state
+     */
+    public double prob() { return Math.exp(sumLogPOfSubgraph); }
```

Comment:
This is going to be very numerically unstable. Most interesting networks are going to return 0 here.

Reply:
How do you reckon we should go about dealing with this? We could remove this functionality for the moment and then raise a ticket to fix this problem?